bob-common/src/metrics/exporters/global_exporter/rate_processor.rs

```diff
@@ -0,0 +1,65 @@
+const BUFFER_SIZE: usize = 10_485_760; // 10 Mb
+use metrics::Key;
+use std::time::Duration;
+use std::{collections::HashMap, time::Instant};
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+
+const RATE_INTERVAL: Duration = Duration::from_secs(5);
+
+type TimeStamp = Instant;
+
+pub(super) struct RateProcessor {
+    tx: Sender<(String, u64, TimeStamp)>,
+}
+
+impl RateProcessor {
+    pub(super) fn run_task() -> Self {
+        let (tx, rx) = channel(BUFFER_SIZE);
+        tokio::spawn(rate_processor_task(rx));
+        Self { tx }
+    }
+
+    pub(super) fn process(&self, key: &Key, value: u64) {
+        if let Err(e) = self
+            .tx
+            .try_send((key.name().to_owned(), value, Instant::now()))
```

Comment:
Why you don't use async methods for channel here?

Reply:
This method is called from sync function (increment_counter), which in its turn defined in trait from metrics crate (I can create runtime and store it somewhere in global exporter, but there would be an overload; don't think, that's good solution)